Assets/Scripts/Controllers/HistoryController.cs

```diff
@@ -21,24 +59,163 @@ public class HistoryController : MonoBehaviour {
+				routesTiles[i] = Instantiate(RouteTemplate, RouteParent.transform);
+				string getFileName = Path.GetFileNameWithoutExtension(name);
+				routesTiles[i].transform.GetChild(0).GetComponent<Text>().text = getFileName;
+				routesTiles[i].transform.GetChild(1).GetComponent<Text>().text = "Route Length: <color=black>" + lengths[i].ToString() + "</color>";
+				routesTiles[i].transform.GetChild(2).name = i.ToString();
+				routesTiles[i].transform.GetChild(2).GetComponent<Button>().onClick.AddListener(() => onRouteButtonClicked());
+				routesTiles[i].transform.position = routesTiles[i].transform.position + new Vector3(0, -64 * i, 0);
+				i++;
+			}
+		}
+
+		public void onRouteButtonClicked() {
+			if (HistoryService.isPlayingRoute()) onRouteExit();
+			int newIndex;
+			if (Int32.TryParse(EventSystem.current.currentSelectedGameObject.name, out newIndex)) {
+				if (newIndex != routeIndex) {
+					routeIndex = newIndex;
+					HistoryService.startRoute(routeIndex);
+					routesTiles[routeIndex].transform.GetComponent<Image>().color = new Color(0.341f, 0.58f, 0.808f, 1.0f);
+					routesTiles[routeIndex].transform.GetChild(2).GetComponent<Button>().transform.GetChild(0).GetComponent<Text>().text = "Stop";
+				}
+				else {
+					routeIndex = -1;
+				}
+
 			}
 		}
+
+		public void makeDefaultColorOnRouteTile() {
+			routesTiles[routeIndex].transform.GetComponent<Image>().color = new Color(0.91f, 0.91f, 0.91f, 0.404f);
+			routesTiles[routeIndex].transform.GetChild(2).GetComponent<Button>().transform.GetChild(0).GetComponent<Text>().text = "Start";
+		}
+
+		public void onRouteExit() {
+			makeDefaultColorOnRouteTile();
+			StopCoroutine(autoRouteCoroutine);
+			HistoryService.stopPlayingRoute();
+		}
+		#endregion
+
+		#region SearchHandling
+		public void createSearchObjects(long index) {
+
+			deleteAllSearchEntries();
+			Dictionary<uint, string> searchResults = HistoryService.searchLoader.getEntries(index);
+			int i = 0;
+			foreach (var result in searchResults) {
+				Node node = nodeController.NodeLoadManager.LoadNode(result.Key);
+				if (node.Type == NodeType.ARTICLE) {
+					searchTiles.Add(Instantiate(SearchTemplateArticle, SearchParent.transform));
+				}
+				else {
+					searchTiles.Add(Instantiate(SearchTemplateCategory, SearchParent.transform));
+				}
+				var t = searchTiles[i].transform.GetChild(1).GetComponent<Text>().text = result.Value.Replace("_", " ");
+				searchTiles[i].GetComponent<Button>().onClick.AddListener(() => OnSearchEntryClicked());
+				searchTiles[i].name = result.Key.ToString();
+				i++;
+
+			}
+
+		}
+
+		private void deleteAllSearchEntries() {
+			foreach (var entry in searchTiles) {
+				Destroy(entry);
+			}
+			searchTiles.Clear();
+		}
+
+		public void OnSearchEntryClicked() {
+			uint index;
+			if (uint.TryParse(EventSystem.current.currentSelectedGameObject.name, out index)) {
+				nodeChangedSource = NodeChangedSource.Search;
+				Node node = nodeController.NodeLoadManager.LoadNode(index);
```

Comment:
Right after this line a force sync is required, to load the node before it is to be selected on clients
```suggestion
				Node node = nodeController.NodeLoadManager.LoadNode(index);
				nodeController.OnNodeLoadSessionEnded?.Invoke();
```

Reply:
Maybe extract those 3 lines to a private function